ddl/ddl_api.go

```diff
@@ -4219,30 +4219,34 @@ func (d *ddl) TruncateTablePartition(ctx sessionctx.Context, ident ast.Ident, sp
 		return errors.Trace(dbterror.ErrPartitionMgmtOnNonpartitioned)
 	}
 
-	var pids []int64
-	if spec.OnAllPartitions {
-		pids = make([]int64, len(meta.GetPartitionInfo().Definitions))
-		for i, def := range meta.GetPartitionInfo().Definitions {
-			pids[i] = def.ID
+	fn := func(pi *model.PartitionInfo) (*model.PartitionInfo, error) {
```

Comment:
Are there anywhere else this helper function would be of use? I.e. getting a PartitionInfo of the new partitions.

Reply:
@mjonss I am not sure about it.  The fn use spec as a argument，it may be not suitable to change it as a common function. If somewhere use it also, i think it can be adjusted at that case.